packages/core/test/unit/query-generator/add-index-query.test.ts

```diff
@@ -0,0 +1,282 @@
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['ifNotExists']),
+      mariadb: 'ALTER TABLE `table` ADD INDEX IF NOT EXISTS `index_1` (`column1`, `column2`)',
+      snowflake: notSupportedError,
+      'postgres sqlite': 'CREATE INDEX IF NOT EXISTS [index_1] ON [table] ([column1], [column2])',
+    });
+  });
+
+  it('generates a query that adds an index with included columns', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', include: ['column3'] }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['include']),
+      db2: new Error('DB2 does not support non-unique indexes with INCLUDE syntax.'),
+      snowflake: notSupportedError,
+      'mssql postgres': 'CREATE INDEX [index_1] ON [table] ([column1], [column2]) INCLUDE ([column3])',
+    });
+  });
+
+  it('generates a query that adds a unique index with included columns', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', include: ['column3'], unique: true }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['include']),
+      snowflake: notSupportedError,
+      'db2 mssql postgres': 'CREATE UNIQUE INDEX [index_1] ON [table] ([column1], [column2]) INCLUDE ([column3])',
+    });
+  });
+
+  it('generates a query that adds a unique index with included columns using an array of liternals', () => {
+    expectsql(() => queryGenerator.addIndexQuery('User', { fields: ['email'], name: 'index_1', include: [sequelize.literal('first_name'), sequelize.literal('last_name')], unique: true }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['include']),
+      snowflake: notSupportedError,
+      'db2 mssql postgres': 'CREATE UNIQUE INDEX [index_1] ON [User] ([email]) INCLUDE (first_name, last_name)',
+    });
+  });
+
+  it('generates a query that adds a SPATIAL index from indexType', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], name: 'index_1', indexType: 'SPATIAL' }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['indexType']),
+      mssql: 'CREATE SPATIAL INDEX [index_1] ON [table] ([column1], [column2])',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD SPATIAL INDEX `index_1` (`column1`, `column2`)',
+    });
+  });
+
+  it('generates a query that adds a index with operator', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'], operator: 'inet_ops' }), {
+      default: buildInvalidOptionReceivedError('addIndexQuery', dialect.name, ['operator']),
+      postgres: 'CREATE INDEX "table_column1_column2" ON "table" ("column1" inet_ops, "column2" inet_ops)',
```

Comment:
what happens if you specify both a column-level operator and a global operator?

Reply:
Added test now